### PR TITLE
Some test cases from Trigoda

### PR DESCRIPTION
**Pull request by TrigodaUG**
I wrote 2 test cases for exercise:

rename-media.spec.ts - changes a media file name in the admin area

countartist.spec.ts - count the number of artist on the front page an compare it with a fixture for there defined amount

### CODE REVIEW

**Comment by nevercodealone** [comment]
Very good sorry for late response i will check now

File: cypress/integration/admin/rename-media.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    loginPage.visit();
+    cy.intercept('/admin/login').as('login');
+    loginPage.doLoginAsAdmin();
+    cy.wait('@login');
+    cy.getCookie('SULUSESSID').should('exist');
+  });
+
+  it('Navigate to media page, find media name field and set new name', () => {
+    adminPage.visit('/admin/#/collections/en');
+    cy.get('.masonry--AWozSpqC-B > li:first .media-card--D92EE5mvsd > .media--ejbY2jJ4HC > .cover--raBSTujlbG > .su-pen').click();
```
Comment (nevercodealone) [inline cypress/integration/admin/rename-media.spec.ts:14]: Ids can break. Maybe better use ".first()" on aria-label="su-pen" 

cy.get(`[aria-label="su-pen"]`).first()

File: cypress/integration/admin/rename-media.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    renameMedia.EditField.click();
+    renameMedia.EditField.clear();
```
Comment (nevercodealone) [inline cypress/integration/admin/rename-media.spec.ts:16]: Please chain methods .click().clear()....

File: cypress/integration/admin/rename-media.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    renameMedia.EditField.type('Wolfsrudel.jpg');
+    renameMedia.EditField.clear();
+    renameMedia.EditField.type('Wolf.jpg');
+    cy.get('.su-save').click();
+    cy.wait(500);
```
Comment (nevercodealone) [inline cypress/integration/admin/rename-media.spec.ts:21]: Wait for response and not for hard coded time. See example on login route

File: cypress/integration/artists/countartists.spec.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    context('EN: artist page', () => {
+
+        it('Load page', () => {
+            cy.visit(`${Cypress.env('URL')}/en/artists`);
+        });
+
+        it('Count and compare with fixture data', function(){
+            expect(this.ArtistCounter.artistsCount.en).to.exist;
+            cy.get('.card').should('have.length', this.ArtistCounter.artistsCount.en).then(_cards=>{
```
Comment (nevercodealone) [inline cypress/integration/artists/countartists.spec.ts:18]: When you can please always use should and avoid callback constructs.

File: cypress/support/pages/rename-media.page.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import BasePage from './base.page';
+
+export class RenameMediaPage extends BasePage {
+  constructor() {
+    super();
+    this.path = '/admin/';
```
Comment (nevercodealone) [inline cypress/support/pages/rename-media.page.ts:6]: There is a admin page with this path. Do we need a new one?

**Review by nevercodealone — CHANGES_REQUESTED** [review]
Hi my friend,

here are some feedback points. I hope you get them. When you have questions please ask.

Thx for your passion
Roland

**Comment by TrigodaUG** [comment]
Hello,

i tried to resolve the issues. Hope it is better now. One thing, yesterday, i could not finish, because the admin account on sulu.rocks didn't work any more. 

Kind regards
Elmar

File: cypress/integration/admin/rename-media.spec.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    cy.intercept('/admin/login').as('login');
+    loginPage.doLoginAsAdmin();
+    cy.wait('@login');
+    cy.getCookie('SULUSESSID').should('exist');
+  });
+
+  it('Navigate to media page, find media name field and set new name', () => {
+    adminPage.visit('/admin/#/collections/en');
+    cy.get('[aria-label="su-pen"]').first().click();
+    cy.get('#\\/title').click().clear().type('Wolfsrudel.jpg').clear().type('Wolf.jpg');
```
Comment (nevercodealone) [inline cypress/integration/admin/rename-media.spec.ts:15]: Why is here a double change?

File: cypress/integration/artists/countartists.spec.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    });
+
+    context('EN: artist page', () => {
+
+        it('Load page', () => {
+            cy.visit(`${Cypress.env('URL')}/en/artists`);
+        });
+
+        it('Count and compare with fixture data', function(){
+            cy.wrap(this.ArtistCounter.artistsCount.en).should('exist').and('gt',0);
```
Comment (nevercodealone) [inline cypress/integration/artists/countartists.spec.ts:17]: should have length is enough. The test will wait for the element. Its better to use Cypress functions in a simple way

File: cypress/integration/artists/countartists.spec.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+            cy.get('.card').should('have.length', this.ArtistCounter.artistsCount.en);
```
Comment (nevercodealone) [inline cypress/integration/artists/countartists.spec.ts:18]: Why hard coded on 5. I think its better to check that there are artists

File: cypress/integration/artists/countartists.spec.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        });
+    });
+
+    context('DE: Musiker Seite', () => {
+        it('Load page', () => {
+            cy.visit(`${Cypress.env('URL')}/de/musiker`);
+        });
+
+        it('Count and compare with fixture data', function(){
+            cy.wrap(this.ArtistCounter.artistsCount.de).should('exist').and('gt',0);
```
Comment (nevercodealone) [inline cypress/integration/artists/countartists.spec.ts:29]: simplify

**Review by nevercodealone — CHANGES_REQUESTED** [review]

**Comment by TrigodaUG** [comment]
Hello,
i got that point and simplyfied it. 

Regarding the hard coded values in the fixture file, my idea was, to fill that with live data in a real world application, which could be a good idea to make testing it more dynamic?

The double fill was, because i first checked the page in Firefox, where ist disabled the save button, when nothing was changed. But it was not nessary in the test, so i deleted them.

Kind regards
Elmar

**Review by nevercodealone — COMMENTED** [review]
Thx

**Pull request merged**